### DISCUSSION TITLE
High cpu usage when dragging windows around

### DISCUSSION THREAD

**Discussion by N3xusPr1me** [rxhanson/Rectangle#475] [Q&A]
_Rectangle 0.46 (51)_

Hello,

I am noticing high CPU usage on my M1 MacBook Air (everything up to date) when dragging windows around, without triggering any snaps by touching the sides with the cursor. 

I have evaluated the load to be approximately an additional 4% on the CPU when continuously dragging. Rectangle has the same activity monitor energy impact as the program I am dragging around ! (~20 for safari and ~20 for Rectangle)

Is this normal behavior because the program always analyzes the mouse's movements ?

Thanks and keep it up

Edit : the high CPU load is fixed when disabling drag to resize - so it has to do with it

**Comment by rxhanson** [comment]
> Is this normal behavior because the program always analyzes the mouse's movements ?

Yes. Rectangle needs to know if the cursor reaches the edge of the display while dragging. Listening to the drag events causes the ~4% load and there's no way around it. Despite the CPU load, the energy usage of Rectangle is still considered low by Apple's metrics. From my own metrics, the energy consumption is negligible in comparison to most applications I run.
